### PR TITLE
Created an option to add password after pdf creation.

### PR DESCRIPTION
**Pull request by shiv07tiwari**
# Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.
If there are any UI change, please include the screenshots also.

Fixes #(issue)

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)

# Checklist:
- [x ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -22,6 +22,7 @@
         <item>Print File</item>
         <item>Share File</item>
         <item>Show Details</item>
+        <item>Add Password</item>
```
Comment (Swati4star) [inline app/src/main/res/values/strings.xml]: Add this also to other strings.xml files

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 2]
```diff
@@ -324,6 +341,67 @@ public String string(@StringRes int resId) {
...
+                Snackbar.make(Objects.requireNonNull(mActivity).findViewById(android.R.id.content),
+                        R.string.password_added,
+                        Snackbar.LENGTH_LONG).show();
+            }
+        });
+
+    }
+    private void doEncryption(String path) throws IOException, DocumentException {
+        PdfReader reader = new PdfReader(path);
+        PdfStamper stamper = new PdfStamper(reader,new FileOutputStream("/storage/emulated/0/PDFfiles/sample.pdf"));
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: Why are you using this constant path

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 3]
```diff
@@ -324,6 +341,67 @@ public String string(@StringRes int resId) {
...
+        stamper.setEncryption(mPassword.getBytes(),"PDF Converter".getBytes(), PdfWriter.ALLOW_PRINTING | PdfWriter.ALLOW_COPY, PdfWriter.ENCRYPTION_AES_256);
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: use `getString(R.string.app_name)`

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 4]
```diff
@@ -324,6 +341,67 @@ public String string(@StringRes int resId) {
...
+        stamper.close();
+        reader.close();
+        File f = new File(path);
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: rename `f` to something else

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 5]
```diff
@@ -324,6 +341,67 @@ public String string(@StringRes int resId) {
             ButterKnife.bind(this, itemView);
         }
     }
+    private void setPassword(final String fileName) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: Add comment to this function

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 6]
```diff
@@ -324,6 +341,67 @@ public String string(@StringRes int resId) {
...
+                dialog.dismiss();
+                notifyDataSetChanged();
+                Snackbar.make(Objects.requireNonNull(mActivity).findViewById(android.R.id.content),
+                        R.string.password_added,
+                        Snackbar.LENGTH_LONG).show();
+            }
+        });
+
+    }
+    private void doEncryption(String path) throws IOException, DocumentException {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: Add comments to this function

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java (hunk 2 above)
Comment (shiv07tiwari) [reply app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: If I delete the file, then how the reader and stamper are gonna encrypt the file.. i guess then it is same as creating another pdf with encryption.

Comment (shiv07tiwari) [reply app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: I am creating an OutputFile from the file to be encrypted.

Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: Then, do this:
Suppose the user wants to encrypt `abc.pdf` file, then create the encrypted version of the file with name `abc_encrypted.pdf`.
Now, the user will have both. He can anyway rename the file 👍

Comment (shiv07tiwari) [reply app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: Okay, so the PDF View is gonna display both encrypted as well as normal one, or should I delete the non encrypted one ?

Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: Don't delete. Let's keep both.

Comment (shiv07tiwari) [reply app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: Okay Sure

**Comment by shiv07tiwari** [comment]
@Swati4star  Made the changes.

**Comment by Swati4star** [comment]
@shiv07tiwari Fix the checkstyle errors.

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 7]
```diff
@@ -324,6 +337,87 @@ public String string(@StringRes int resId) {
...
+        final String[] name_withoutpdf = filename.split("\\.");
+
+        String filename_withoutpdf = name_withoutpdf[0] + "_encrypted.pdf";
+        String final_output_file = final_path + filename_withoutpdf;
+        PdfReader reader = new PdfReader(path);
+        PdfStamper stamper = new PdfStamper(reader,new FileOutputStream(final_output_file));
+        stamper.setEncryption(mPassword.getBytes(),mActivity.getString(R.string.app_name).getBytes(), PdfWriter.ALLOW_PRINTING | PdfWriter.ALLOW_COPY, PdfWriter.ENCRYPTION_AES_256);
+        stamper.close();
+        reader.close();
+    }
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: Add `mFileList.add(new File(final_output_file));` here & everything will work fine.

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 8]
```diff
@@ -324,6 +337,87 @@ public String string(@StringRes int resId) {
...
+    private void doEncryption(String path) throws IOException, DocumentException {
+        final String[] name_withpdf = path.split("/");
+        String final_path = "";
+        String filename= name_withpdf[name_withpdf.length - 1];
+        for(int i=0;i<name_withpdf.length-1;i++)
+        {
+            final_path = final_path + name_withpdf[i]+"/";
+        }
+
+        final String[] name_withoutpdf = filename.split("\\.");
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: You have quite messed up the logic here.

Why are you splitting it by `/`
Just remove the `.pdf` from `path` & replace it with `_encrypted.pdf`

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 9]
```diff
@@ -324,6 +337,87 @@ public String string(@StringRes int resId) {
...
+
+        String filename_withoutpdf = name_withoutpdf[0] + "_encrypted.pdf";
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: Add `_encrpted.pdf` to `strings.xml` & use that here

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 10]
```diff
@@ -324,6 +337,87 @@ public String string(@StringRes int resId) {
...
+        String final_output_file = final_path + filename_withoutpdf;
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: Use camel case for variable names

**Comment by shiv07tiwari** [comment]
@Swati4star  Made all the changes with checksyle fix.

**Comment by shiv07tiwari** [comment]
Done

**Pull request merged**